### DISCUSSION TITLE
Unexpected line connect from node to other node

### DISCUSSION THREAD

**Discussion by doanhoangvu98** [xyflow/xyflow#4024] [General]
![image](https://github.com/xyflow/xyflow/assets/42468423/c7cef99f-ffc9-4a4a-b2c5-a655f578d026)
![image](https://github.com/xyflow/xyflow/assets/42468423/402c508f-20c8-45f2-a700-96da8ab06770)

Hi everyone, I'm stuck when connecting between node. Anyone if know connects from completed to finish with a straight line like the second picture please help me. Here is my code. I'm appreciate if anyone could help. Thanks

`import { useCallback } from 'react';
import ReactFlow, { MarkerType, Node, Position, addEdge, useEdgesState, useNodesState } from 'reactflow';
import 'reactflow/dist/style.css';
import customNode from './custom-node';

const nodeTypes = {
  custom: customNode,
};

const initialNodes: Node<any, string | undefined>[] = [
  {
    id: 'horizontal-1',
    sourcePosition: Position.Right,
    // targetPosition: Position.Right,
    type: 'custom',
    data: { label: 'Initial' },
    position: { x: 0, y: 0 },
    draggable: false,
  },
  {
    id: 'horizontal-2',
    type: 'custom',
    sourcePosition: Position.Right,
    targetPosition: Position.Left,
    data: { label: 'Sent' },
    position: { x: 200, y: 0 },
    draggable: false,
  },
  {
    id: 'horizontal-3',
    type: 'custom',
    sourcePosition: Position.Top,
    targetPosition: Position.Left,
    data: { label: 'Waiting for other' },
    position: { x: 400, y: 0 },
    draggable: false,
  },
  {
    id: 'horizontal-4',
    type: 'custom',
    sourcePosition: Position.Right,
    targetPosition: Position.Left,
    data: { label: 'Completed' },
    position: { x: 600, y: 0 },
    draggable: false,
  },
  {
    id: 'horizontal-5',
    type: 'custom',
    sourcePosition: Position.Right,
    targetPosition: Position.Top,
    // targetPosition: Position.Left,
    data: { label: 'Finish' },
    position: { x: 800, y: 0 },
    draggable: false,
  },
  {
    id: 'horizontal-6',
    type: 'custom',
    sourcePosition: Position.Right,
    targetPosition: Position.Top,
    data: { label: 'Custom drag drop zone' },
    position: { x: 400, y: 100 },
    draggable: false,
  },
  {
    id: 'horizontal-7',
    type: 'custom',
    sourcePosition: Position.Right,
    targetPosition: Position.Left,
    data: { label: 'Voided' },
    position: { x: 600, y: -100 },
    draggable: false,
  },
];

const initialEdges = [
  {
    id: 'horizontal-e1-2',
    source: 'horizontal-1',
    type: 'smoothstep',
    target: 'horizontal-2',
    // animated: true,
    markerEnd: {
      type: MarkerType.Arrow,
    },
  },
  {
    id: 'horizontal-e1-3',
    source: 'horizontal-1',
    type: 'smoothstep',
    target: 'horizontal-3',
    // animated: true,
    markerEnd: {
      type: MarkerType.Arrow,
    },
  },
  {
    id: 'horizontal-e1-4',
    source: 'horizontal-2',
    type: 'smoothstep',
    target: 'horizontal-4',
    label: 'edge label',
    markerEnd: {
      type: MarkerType.Arrow,
    },
  },
  {
    id: 'horizontal-e3-6',
    source: 'horizontal-3',
    type: 'smoothstep',
    target: 'horizontal-7',
    // animated: true,
    markerEnd: {
      type: MarkerType.Arrow,
    },
  },

  {
    id: 'horizontal-e3-7',
    source: 'horizontal-7',
    type: 'smoothstep',
    target: 'horizontal-5',
    markerEnd: {
      type: MarkerType.Arrow,
    },
    // animated: true,
  },
  {
    id: 'horizontal-e3-8',
    source: 'horizontal-4',
    type: 'smoothstep',
    target: 'horizontal-5',
    markerEnd: {
      type: MarkerType.Arrow,
    },
    // animated: true,
  },
];

const DiagramFlow = () => {
  const [nodes, _, onNodesChange] = useNodesState(initialNodes);
  const [edges, setEdges, onEdgesChange] = useEdgesState(initialEdges);
  const onConnect = useCallback((params: any) => setEdges((els) => addEdge(params, els)), []);

  return (
    <div className="w-full h-full border-[1px] border-gray-900 m-3">
      <ReactFlow
        nodes={nodes}
        edges={edges}
        onNodesChange={onNodesChange}
        onEdgesChange={onEdgesChange}
        onConnect={onConnect}
        fitView
        // panOnDrag={false}
        // zoomOnScroll={false}
        // nodeTypes={nodeTypes}
        attributionPosition="bottom-left"
      ></ReactFlow>
    </div>
  );
};

export default DiagramFlow;
`

**Comment by bcakmakoglu** [comment]
Not sure what the issue is exactly - you want it to connect on the left side but you have this on your code

```tsx
targetPosition: Position.Top,
// targetPosition: Position.Left,
```

Why comment out `targetPosition: Position.Left` when that's apparently what you want in the first place? 😅
